jupiter-tests/src/test/java/org/junit/jupiter/testkit/ExecutionResults.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.testkit;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.StreamSupport;
+
+import org.junit.platform.engine.DiscoverySelector;
+import org.junit.platform.engine.discovery.DiscoverySelectors;
+import org.junit.platform.testkit.engine.EngineExecutionResults;
+import org.junit.platform.testkit.engine.EngineTestKit;
+import org.junit.platform.testkit.engine.Events;
+
+/**
+ * Pioneers' class to handle JUnit Jupiter's {@link org.junit.platform.testkit.engine.EngineExecutionResults}.
+ *
+ * <p>Instantiate with the static factory methods in {@link JUnitJupiterTestKit}.
+ */
+public class ExecutionResults {
```

Comment:
I would like to introduce a new mechanism for this. We have a `AbstractJupiterTestEngineTests` base class that should provide a few helper methods that you can use instead.

Reply:
Will have a look on Tuesday or maybe Thursday/Weekend. But feel free to take over